editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx

```diff
@@ -0,0 +1,476 @@
+import { Utils } from '../../../uuiui-deps'
+import { UIFileBase64Blobs } from '../../editor/store/editor-state'
+import { UiJsxCanvasContextData } from '../ui-jsx-canvas'
+import { SceneRoot } from './scene-root'
+import * as PP from '../../../core/shared/property-path'
+import * as TP from '../../../core/shared/template-path'
+import { Storyboard } from 'utopia-api'
+import { resolveParamsAndRunJsCode } from '../../../core/shared/javascript-cache'
+import { objectMap } from '../../../core/shared/object-utils'
+import { cssValueOnlyContainsComments } from '../../../printer-parsers/css/css-parser-utils'
+import { filterDataProps } from '../../../utils/canvas-react-utils'
+import { buildSpyWrappedElement } from './ui-jsx-canvas-spy-wrapper'
+
+export function renderCoreElement(
+  element: JSXElementChild,
+  templatePath: InstancePath,
+  rootScope: MapLike<any>,
+  inScope: MapLike<any>,
+  parentComponentInputProps: MapLike<any>,
+  requireResult: MapLike<any>,
+  hiddenInstances: Array<TemplatePath>,
+  fileBlobs: UIFileBase64Blobs,
+  validPaths: Array<InstancePath>,
+  uid: string | undefined,
+  reactChildren: React.ReactNode | undefined,
+  metadataContext: UiJsxCanvasContextData,
+  jsxFactoryFunctionName: string | null,
+  codeError: Error | null,
+  shouldIncludeCanvasRootInTheSpy: boolean,
+): React.ReactElement {
+  if (codeError != null) {
+    throw codeError
+  }
+  if (isJSXElement(element) && isSceneElement(element)) {
+    const sceneProps = jsxAttributesToProps(
+      inScope,
+      element.props,
+      parentComponentInputProps,
+      requireResult,
+    )
+
+    const rootComponent = sceneProps.component
+    const rootComponentName = sceneProps.component?.topLevelElementName
+    const resizesContent = Boolean(
+      Utils.path(PP.getElements(PathForResizeContent), sceneProps) ?? false,
+    )
+
+    const sceneId: string = sceneProps['data-uid'] || ''
+    return (
+      <SceneRoot
+        content={rootComponent} // this is the child component
+        componentProps={sceneProps.props}
+        container={sceneProps.layout}
+        hiddenInstances={hiddenInstances}
+        jsxFactoryFunctionName={jsxFactoryFunctionName}
+        fileBlobs={fileBlobs}
+        style={sceneProps.style}
+        inScope={inScope}
+        requireResult={requireResult}
+        templatePath={templatePath}
+        component={rootComponentName}
+        sceneResizesContent={resizesContent}
+        sceneUID={sceneId}
+        sceneLabel={sceneProps['data-label']}
+      />
+    )
+  }
+  switch (element.type) {
+    case 'JSX_ELEMENT': {
+      const assembledProps = jsxAttributesToProps(
+        inScope,
+        element.props,
+        parentComponentInputProps,
+        requireResult,
+      )
+
+      let passthroughProps: MapLike<any> = {
+        ...assembledProps,
+        'data-uid': Utils.defaultIfNull(assembledProps['data-uid'], uid),
```

Comment:
These shouldn't be here as they're added in the lines immediately following this.

Reply:
fixed! today I seem to have a problem seeing the difference between the red rows and the green rows :)